### PR TITLE
Fix up editor tooltip

### LINKED ISSUE #4421: Inline Documentation Does Not Render Correclty
**Application or Package Used**
nteract desktop

**Describe the bug**
When I try to look at the documentation using the keyboard shortcut (CTRL + .) the documentation renders, but it is not opaque. Meaning the notebook code blends in with the documentation code. Additionally, the "X" to close the documentation is not clickable, so it cannot be closed.

**To Reproduce**
Launch documentation with keyboard shortcut.

**Expected behaviour**
Documentation launches in a form that is readable.

**Screenshots**
image

**Desktop Version

- OS: 10.14.4 macOS 
- Version: 0.14.3 (0.14.3)

### PR DESCRIPTION
**Pull request by BenRussert**
Changes: 
- Esc destroys tooltip (from editor or tooltip itself)
- Ctrl-. destroys tooltip if already open (toggle)
- Tip position updates based on current cursor position
- Close #4421

### CODE REVIEW

File: packages/editor/src/index.tsx [hunk 1]
```diff
@@ -246,6 +233,18 @@ export default class CodeMirrorEditor extends React.PureComponent<
 
     const { completion, editorFocused, focusAbove, focusBelow } = this.props;
 
+    // ensure a single tooltip holder exists on document.body
+    const tipHolder = document.getElementsByClassName(
+      "tip-holder"
+    )[0] as HTMLDivElement;
```
Comment (rgbkrk) [inline packages/editor/src/index.tsx:239]: Is this guaranteed to return a non-empty array? We probably need the bounds checked on this.

File: packages/editor/src/index.tsx [hunk 2]
```diff
@@ -263,8 +262,16 @@ export default class CodeMirrorEditor extends React.PureComponent<
...
 
-    this.cm.on("topBoundary", focusAbove!);
-    this.cm.on("bottomBoundary", focusBelow!);
+    this.cm.on("topBoundary", (editor: Editor) => {
+      this.deleteTip();
+      focusAbove!(editor);
+    });
+    this.cm.on("bottomBoundary", (editor: Editor) => {
+      this.deleteTip();
+      focusBelow!(editor);
```
Comment (rgbkrk) [inline packages/editor/src/index.tsx]: Is this similar to the `?` operator from coffeescript or is this forcing `focusBelow` to be assumed non-null?

File: packages/editor/src/components/tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    <Tip
+      className="CodeMirror-hint"
+      tabIndex={0}
+      onKeyDown={e => {
+        e.key === "Escape" ? deleteTip() : null;
+      }}
+      cursorCoords={cursorCoords}
+    >
+      <RichMedia data={bundle} metadata={{ expanded: true }}>
+        <Media.Markdown />
```
Comment (BenRussert) [inline packages/editor/src/components/tooltip.tsx:52]: By the way markdown wasn't here before. The IJulia kernel often sends back markdown docs from inspection requests, so we've supported it in Hydrogen for a while. (Julia kernel is another issue #4415, anybody let me know if you want to work on that I've been stumped a couple times now)

File: packages/editor/src/index.tsx (hunk 1 above)
Comment (BenRussert) [reply packages/editor/src/index.tsx:239]: Yep, if the class name ("tip-holder" in this case) returns no results, the result is an empty iterable/array-like `HTMLDivElement`

File: packages/editor/src/index.tsx (hunk 2 above)
Comment (BenRussert) [reply packages/editor/src/index.tsx]: Yes, non-null assertion operator. I carried this over from the previous code. It seems better to remove.

**Pull request merged**